src/esphome/sensor/mpr121_sensor.cpp

```diff
@@ -0,0 +1,90 @@
+#include "esphome/defines.h"
+#ifdef USE_MPR121
+
+#include "esphome/sensor/mpr121_sensor.h"
+#include "esphome/log.h"
+#include <bitset>
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace sensor {
+
+static const char *TAG = "sensor.mpr121";
+
+MPR121SensorChannel::MPR121SensorChannel(uint8_t channel, float value) {
+  this->channel = channel;
+  this->value = value;
+}
+
+MPR121Sensor::MPR121Sensor(const std::string &name) : Sensor(name) {
+  this->state = 0.0;
+ }
+
+void MPR121Sensor::process(uint16_t *data, uint16_t *last_data) {
+  //process depend on the sensor_type_
+  switch(this->sensor_type_) {
+    case MPR121_SENSOR_TYPE_SLIDER:
+      this->process_slider_(data);
+      break;
+    case MPR121_SENSOR_TYPE_ROTARY:
+      this->process_rotary_(data, last_data);
+      break;
+    default:
+      ESP_LOGE(TAG, "'%s': Unknown sensor type %d", this->name_.c_str(), this->sensor_type_);
```

Comment:
Let's just `break;` - Because now esphome should only be used with the python app we don't need to handle this.